src/components/Trends/TrendDepthToggle.js

```diff
@@ -14,63 +18,90 @@ const lensMap = {
   Company: 'company',
 };
 
-export class TrendDepthToggle extends React.Component {
-  _showMore() {
-    const { queryCount, resultCount } = this.props;
-    // scenarios where we want to show more:
-    // you have less visible rows that the max (5)
-    if (resultCount <= maxRows) {
-      return true;
-    }
-    // or more filters count > max Rows and they aren't the same (visible)
-    return queryCount > maxRows && queryCount !== resultCount;
+const _showMore = (filterCount, resultCount) => {
+  // scenarios where we want to show more:
+  // you have less visible rows that the max (5)
+  if (resultCount <= maxRows) {
+    return true;
   }
+  // or more filters count > max Rows and they aren't the same (visible)
+  return filterCount > maxRows && filterCount !== resultCount;
+};
 
-  render() {
-    const { diff, increaseDepth, depthReset, hasToggle } = this.props;
-    if (hasToggle) {
-      if (this._showMore()) {
-        return (
-          <div className="trend-depth-toggle">
-            <button
-              className="a-btn a-btn--link"
-              id="trend-depth-button"
-              onClick={() => {
-                increaseDepth(diff);
-              }}
-            >
-              <span className="plus" />
-              Show more
-            </button>
-          </div>
-        );
-      }
+export const TrendDepthToggle = () => {
+  const dispatch = useDispatch();
+  const aggs = useSelector(selectAggsState);
+  const query = useSelector(selectQueryState);
+  const focus = useSelector(selectQueryFocus);
+  const lens = useSelector(selectQueryLens);
+  const results = useSelector(selectTrendsResults);
+  const lensKey = lensMap[lens];
+  const resultCount = coalesce(results, lensKey, []).filter(
+    (obj) => obj.isParent,
+  ).length;
+
+  // The total source depends on the lens.  There are no aggs for companies
+  let totalResultsLength = 0;
+  if (lensKey === 'product') {
+    totalResultsLength = coalesce(aggs, lensKey, []).length;
+  } else {
+    totalResultsLength = clamp(coalesce(query, lensKey, []).length, 0, 10);
+  }
+
+  // handle cases where some specified filters are selected
+  const queryCount = query[lensKey]
+    ? query[lensKey].filter((obj) => obj.indexOf(SLUG_SEPARATOR) === -1).length
+    : totalResultsLength;
+
+  const diff = totalResultsLength - resultCount;
+  const hasToggle = showToggle(lens, focus, totalResultsLength, queryCount);
+
+  // hide on Overview and Focus pages
+  if (focus || lens === 'Overview') {
+    return null;
+  }
+
+  if (hasToggle) {
+    if (_showMore(queryCount, resultCount)) {
```

Comment:
Not a big deal, but would the underscore at the start of `_hasMore` be necessary here? I assumed they were present as a way to indicate they were on the instance of the TrendDepthToggle class component.

Reply:
No.  I think someone else did the underscore for functions in the past as private functions.  

Normally, I would not use them.  I'll remove them.